Opencast-Moodle/moodle-block_opencast
# Autocomplete Suggestions using capability

This PR fixes #302,

### Description
The autocomplete suggestions for contributor and presentor fields consists of 3 different part, one of them is to extract the teachers name enrolled in the course, this process had a few problems along the way, one was that it was using shortname "editingteacher" to get the role id which was incorrect as it is a parameter that could be changed, and for those institutes that have changed this parameter, this process throws error and does not work properly.

### Solution
A new capability is introduced to provide appropriate roles defined by "archetypes". And on the other hands the process will now use a built-in method "get_enrolled_users" from Moodle to get the enrolled users who have that capability.

### To Test and run this PR
- An upgrade to a new version of this plugin is needed (Automatically). Because a new capability is introduced in `access.php` file, therefore, to test/run this PR one should pay attention to this version bump, to avoid versioning confilct etc.
- Make sure the test course have a few teachers enrolled.
- In update metadata / add video / series add or update pages you have be able to see the teachers' fullnames as autocomplete suggestions dropdowns for Presenter and Contributor fields.
